Automattic/jetpack
# Boost: Update source providers collection logic

Fixes #37090

## Proposed changes:

* Add ignore list for post type providers (currently only includes `element-hf` from https://wordpress.org/plugins/header-footer-elementor/;
* Add ignore list for archive post type providers;
* Fix incorrectly checking if archive post types, taxonomies and post type providers are viewable;
* Remove home/blog page from the list of singular provider URLs. It doesn't make sense to have them for separate groups.
  

<details>
<summary>Home and blog previously in the singular_page list:</summary>

![image](https://github.com/Automattic/jetpack/assets/11799079/bf7d7340-f239-46f4-9abe-0d0b9c9ad552)

</details>

<details>
<summary>Home and blog no longer in the singular_page list:</summary>

![CleanShot 2024-05-20 at 17 16 31](https://github.com/Automattic/jetpack/assets/11799079/d6f6fb19-a069-43c7-ab01-c0381c1b796b)

</details>

### Other information:


## Jetpack product discussion



pc9hqz-2P1-p2#comment-2091

## Does this pull request change what data or activity we track or use?



no

## Testing instructions:






> [!NOTE]
> The easiest way to test this is to access the source providers endpoint that is now available in Boost (http://jetpack-boost.test/wp-json/jetpack-boost/v1/list-source-providers). However, since it's only accessible with a blog token, you need to be running the [Boost Developer plugin](https://github.com/Automattic/boost-developer/), so make sure you have that enabled first.

- Install and setup elementor (you can do that by grabbing the version from the [github repo](https://github.com/elementor/elementor/releases));
- Install and setup https://wordpress.org/plugins/header-footer-elementor/;
- Add some header/footer sections, by going to Appearance -> Elementor Header & Footer Builder;
- Make sure those sections don't show up in the endpoint.

You can also register a custom endpoint and play around with the args to make sure things behave appropriately. You can enable/disable its archive page, make it publicly queryable, etc.

Here's an example custom post type with an archive page:

```php
function pp_register_post_type() {
	register_post_type( 'my-post-type', array(
		'public' => true,
		'label'  => 'My Post Type',
		'has_archive' => 'mpt',
	) );
}

add_action( 'init', 'pp_register_post_type' );
```

Just make sure to refresh your permalinks after you add this.
